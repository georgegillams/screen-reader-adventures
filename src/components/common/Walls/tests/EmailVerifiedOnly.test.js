import React from 'react';
import { render } from '@testing-library/react';

import { EmailVerifiedOnly } from '..';

describe('<EmailVerifiedOnly />', () => {
  it('Should render correctly', () => {
    const { container } = render(<EmailVerifiedOnly user={null}>Test</EmailVerifiedOnly>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with user', () => {
    const { container } = render(<EmailVerifiedOnly user={{}}>Test</EmailVerifiedOnly>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with user with verified email', () => {
    const { container } = render(<EmailVerifiedOnly user={{ emailVerified: true }}>Test</EmailVerifiedOnly>);

    expect(container).toMatchSnapshot();
  });
});
