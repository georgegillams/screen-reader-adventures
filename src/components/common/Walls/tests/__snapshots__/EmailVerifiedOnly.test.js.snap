// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailVerifiedOnly /> Should render correctly 1`] = `
<div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="you-need-to-verify-your-email-before-completing-this-step."
    >
      You need to verify your email before completing this step.
    </h2>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline"
      href="/account"
    >
      Account
    </a>
  </div>
</div>
`;

exports[`<EmailVerifiedOnly /> Should render correctly with user 1`] = `
<div>
  <div
    class="section__outer"
  >
    <h2
      class="section__heading"
      id="you-need-to-verify-your-email-before-completing-this-step."
    >
      You need to verify your email before completing this step.
    </h2>
    <a
      class="text-link text-link--text-link text-link--no-padding text-link--inline"
      href="/account"
    >
      Account
    </a>
  </div>
</div>
`;

exports[`<EmailVerifiedOnly /> Should render correctly with user with verified email 1`] = `
<div>
  Test
</div>
`;
